library/src/blas1/rocblas_amax.cpp

```diff
@@ -206,22 +204,26 @@ rocblas_amax_template(rocblas_handle handle,
 
     rocblas_status status;
 
-    T2 *workspace;
-    rocblas_int *workspace_index;
+    auto workspace_managed = rocblas_unique_ptr{rocblas::device_malloc(sizeof(T2) * blocks),rocblas::device_free};
+    T2 *workspace = (T2 *) workspace_managed.get();
```

Comment:
I'm pretty sure you can treat an unique_ptr just like a normal pointer, i.e. there is no need to call the .get() method to retrieve the raw pointer.  You should be able to use the ! operator on the unique_ptr and never touch the raw pointer